mil_common/drivers/mil_passive_sonar/scripts/ping_publisher.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+import contextlib
+import json
+import socket
+
+import rospy
+from geometry_msgs.msg import Point
+from mil_passive_sonar.msg import ProcessedPing
+from std_msgs.msg import Header
+
+
+def main():
+    # Define the server address and port
+    HOST = "127.0.0.1"
+    PORT = 2007  # The port used by the server
+
+    # Create a socket
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.connect((HOST, PORT))
+        print(f"Connected to {HOST}:{PORT}")
+        pub = rospy.Publisher("ping", ProcessedPing, queue_size=10)
+        rospy.init_node("pingpublisher", anonymous=True)
+
+        while not rospy.is_shutdown():
+            # Receive data
+            data = s.recv(1024)  # Buffer size is 1024 bytes
+            if not data:
+                break
+
+            # Parse the JSON data
+            try:
+                json_data = json.loads(data.decode("utf-8"))
+                # Create a ProcessedPing message
+                ping_msg = ProcessedPing()
+
+                # Populate the header
+                ping_msg.header = Header()
+                ping_msg.header.seq += 1  # Increment sequence number
+                ping_msg.header.stamp = rospy.Time.now()
+                ping_msg.header.frame_id = "ping_frame"  # Set your frame_id
```

Comment:
The [frame name should be `hydrophones`](https://github.com/uf-mil/mil/blob/master/NaviGator/simulation/navigator_gazebo/urdf/navigator.urdf.xacro#L25)